codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/RustClientCodegenPlugin.kt

```diff
@@ -72,6 +73,7 @@ class RustClientCodegenPlugin : ClientDecoratableBuildPlugin() {
                 IdempotencyTokenDecorator(),
                 StalledStreamProtectionDecorator(),
                 StaticSdkFeatureTrackerDecorator(),
+                ClientSerdeDecorator(),
```

Comment:
so this turns it on by default for all clients? I'm a little worried this would accidentally start generating serializers for the Rust SDK if a team ever exported the `serde` trait in their model. I think we need to make this opt-in in some way, e.g. via a codegen setting if we are going to bundle it like this.